static/app/components/events/viewHierarchy/wireframe.tsx

```diff
@@ -1,62 +1,93 @@
-import {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {useCallback, useEffect, useLayoutEffect, useMemo, useState} from 'react';
+import {useTheme} from '@emotion/react';
 import styled from '@emotion/styled';
-import {useResizeObserver} from '@react-aria/utils';
+import {mat3, vec2} from 'gl-matrix';
 
 import {ViewHierarchyWindow} from 'sentry/components/events/viewHierarchy';
-import {useMousePan} from 'sentry/components/events/viewHierarchy/useMousePan';
+import {defined} from 'sentry/utils';
+import {Rect, watchForResize} from 'sentry/utils/profiling/gl/utils';
+
+function useResizeCanvasObserver(canvases: (HTMLCanvasElement | null)[]): Rect {
+  const [bounds, setCanvasBounds] = useState<Rect>(Rect.Empty());
+
+  useLayoutEffect(() => {
+    if (!canvases.length) {
+      return undefined;
+    }
+
+    if (canvases.some(c => c === null)) {
+      return undefined;
+    }
+
+    const observer = watchForResize(canvases as HTMLCanvasElement[], entries => {
+      const contentRect =
+        entries[0].contentRect ?? entries[0].target.getBoundingClientRect();
+
+      setCanvasBounds(
+        new Rect(
+          contentRect.x,
+          contentRect.y,
+          contentRect.width * window.devicePixelRatio,
+          contentRect.height * window.devicePixelRatio
+        )
+      );
+    });
+
+    return () => {
+      observer.disconnect();
+    };
+  }, [canvases]);
+
+  return bounds;
+}
 
-const RECT_FILL_ALPHA = 0.005;
-const RECT_OUTLINE_ALPHA = 1;
 const MIN_BORDER_SIZE = 20;
 
-type Rect = Pick<ViewHierarchyWindow, 'x' | 'y' | 'width' | 'height'>;
+interface ViewNode {
+  node: ViewHierarchyWindow;
+  rect: Rect;
+}
```

Comment:
We're returning the node and the corresponding rect here because it'll be used to figure out selection later on